src/OpenRoad.cc

```diff
@@ -456,8 +457,7 @@ void OpenRoad::readDb(const char* filename, bool hierarchy)
   stream.open(filename, std::ios::binary);
 
   try {
-    const std::string name(filename);
-    if (name.length() >= 3 && name.compare(name.length() - 3, 3, ".gz") == 0) {
+    if (boost::ends_with(std::string(filename), ".gz")) {
       boost::iostreams::filtering_streambuf<boost::iostreams::input> inbuf;
       inbuf.push(boost::iostreams::gzip_decompressor());
       inbuf.push(stream);
```

Comment:
Do you think we should do something similar to this for gz read?

Reply:
yeah, we could, just need a class to wrap `ifstream`